lib/instrumentation/modules/redis.js

```diff
@@ -46,37 +49,28 @@ module.exports = function (redis, agent, version) {
 
   function wrapSendCommand (original) {
     return function wrappedSendCommand (command) {
-      var span = agent.buildSpan()
+      var span = enabled && agent.buildSpan()
       var id = span && span.transaction.id
       var args = Array.prototype.slice.call(arguments)
 
       agent.logger.debug('intercepted call to RedisClient.prototype.internal_send_command %o', { id: id, command: command })
 
-      if (span && args.length > 0) {
```

Comment:
Why dropping the check for `span` here? Is it to ensure that the callback is bound even if there's no active transaction?

Reply:
Yep, that's exactly it. I do `var span = enabled && agent.buildSpan()` above, so the span will only exist when in a sampled transaction *and* the redis instrumentation is enabled.